addons/gamification/models/gamification_challenge.py

```diff
@@ -542,12 +542,12 @@ def _get_serialized_challenge_lines(self, user=(), restrict_goals=(), restrict_t
 
             line_data['own_goal_id'] = False,
             line_data['goals'] = []
-            if line.condition=='higher':
-                goals = Goals.search(domain, order="completeness desc, current desc")
-            else:
-                goals = Goals.search(domain, order="completeness desc, current asc")
+            goals = Goals.search(domain)
             if not goals:
                 continue
+            goals = sorted(goals, key=lambda x: (
```

Comment:
You can sort a recordset using the `sorted` method in the `Model` class.
[sorted method](https://github.com/odoo/odoo/blob/3c7db87ade7e99eafb44228bd038fa39429fdabc/odoo/models.py#L6258)